RecastDemo/CMakeLists.txt

```diff
@@ -34,6 +34,22 @@ if(APPLE)
   include_directories(${SDL2_LIBRARY}/Headers)
 endif()
 
+file(GLOB DEBUG_UTILS_SRC "${CMAKE_CURRENT_SOURCE_DIR}/../DebugUtils/Source/*.cpp")
+add_library(DebugUtils SHARED ${DEBUG_UTILS_SRC})
+
+file(GLOB RECAST_SRC "${CMAKE_CURRENT_SOURCE_DIR}/../Recast/Source/*.cpp")
+add_library(Recast SHARED ${RECAST_SRC})
+
+file(GLOB RECAST_DETOUR_SRC "${CMAKE_CURRENT_SOURCE_DIR}/../Detour/Source/*.cpp")
+add_library(Detour SHARED ${RECAST_DETOUR_SRC})
+
+file(GLOB RECAST_DETOUR_CROWD_SRC "${CMAKE_CURRENT_SOURCE_DIR}/../DetourCrowd/Source/*.cpp")
+add_library(DetourCrowd SHARED ${RECAST_DETOUR_CROWD_SRC})
+
+file(GLOB RECAST_DETOUR_TILE_CACHE_SRC "${CMAKE_CURRENT_SOURCE_DIR}/../DetourTileCache/Source/*.cpp")
+add_library(DetourTileCache SHARED ${RECAST_DETOUR_TILE_CACHE_SRC})
```

Comment:
This adds duplicated definitions for libraries built in other places:

* [Recast](https://github.com/recastnavigation/recastnavigation/blob/455a019e7aef99354ac3020f04c1fe3541aa4d19/Recast/CMakeLists.txt#L4)
* [DebugUtils](https://github.com/recastnavigation/recastnavigation/blob/455a019e7aef99354ac3020f04c1fe3541aa4d19/DebugUtils/CMakeLists.txt#L4)
* [Detour](https://github.com/recastnavigation/recastnavigation/blob/455a019e7aef99354ac3020f04c1fe3541aa4d19/Detour/CMakeLists.txt#L4)
* [DetourCrowd](https://github.com/recastnavigation/recastnavigation/blob/455a019e7aef99354ac3020f04c1fe3541aa4d19/DetourCrowd/CMakeLists.txt#L4)
* [DetourTileCache](https://github.com/recastnavigation/recastnavigation/blob/455a019e7aef99354ac3020f04c1fe3541aa4d19/DetourTileCache/CMakeLists.txt#L4)

This is not what should be done here.